### PR TITLE
Add particles sample

### PR DESCRIPTION
**Pull request by ben-clayton**
Demonstrates some basic particle simualation with compute shaders,
generation of probability maps from a loaded image.

### CODE REVIEW

**Comment by ben-clayton** [comment]
Feel free to heavily critique.
This is my first wgpu sample. Actually pretty much my first WebGPU anything, aside from trivial tests.

File: src/sample/particles/main.ts [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+    },
+    fragment: {
+      module: device.createShaderModule({
+        code: particleWGSL,
+      }),
+      entryPoint: 'fs_main',
+      targets: [
+        {
+          format: presentationFormat,
+          blend: {
```
Comment (Kangz) [inline src/sample/particles/main.ts:92]: missing blend op? I'm not sure what's the default

File: src/sample/particles/main.ts [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+  let textureHeight = 1;
+  let numMipLevels = 1;
+  {
+    const img = document.createElement('img');
+    img.src = require('../../../assets/img/webgpu.png');
+    await img.decode();
+    const imageBitmap = await createImageBitmap(img);
+
+    // Calculate number of mip levels required to generate the probability map
+    while (
```
Comment (Kangz) [inline src/sample/particles/main.ts:204]: Can't you directly create a texture for the correct size? WebGPU supports non-power of two textures.

Comment (ben-clayton) [reply src/sample/particles/main.ts:204]: The texture can be non-POT, but we need to generate mipmaps up to 1x1

File: src/sample/particles/main.ts (hunk 1 above)
Comment (ben-clayton) [reply src/sample/particles/main.ts:92]: Dawn seemingly defaults to `'add'`. Added to be explicit.

File: src/sample/particles/probabilityMap.wgsl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  buf_out.weights[offset] = textureLoad(tex_in, vec2<i32>(coord.xy), 0).w;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// export_level
+//
+// Loads 4 f32 weight values from buf_in.weights, and stores summed value into
+// buf_out.weights, along with the calculated 'probabilty' vec4 values into the
+// mip level of tex_out. See simulate() in particle.wgsl to understand the
+// probibility logic.
```
Comment (austinEng) [inline src/sample/particles/probabilityMap.wgsl:35]:
```suggestion
// probability logic.
```

**Review by austinEng — APPROVED** [review]
looks great! sorry for the slow response

**Comment by ben-clayton** [comment]
I do not have permissions to merge this, so if you're happy for this to go in, please feel free to merge.

**Pull request merged**